lib/pbench/server/database/models/datasets.py

```diff
@@ -560,6 +560,17 @@ def attach(path=None, controller=None, name=None, state=None) -> "Dataset":
             dataset.advance(state)
         return dataset
 
+    @staticmethod
+    def query(**kwargs) -> "Dataset":
+        """
+        Query dataset object based on a given column name of the run document
+        """
+        try:
+            dataset = Database.db_session.query(Dataset).filter_by(**kwargs).first()
+            return dataset
+        except Exception:
+            raise
```

Comment:
Don't ever catch an exception just to raise it; that's pointless. This is exactly the same as omitting the `try` entirely, except slower and harder to read.

On the other hand, a log message here might not be bad, and generally I was passing on SQLAlchemy errors using the `DatasetSqlError` exception, which would be implicitly chained from whatever you'd caught here.

Reply:
> generally I was passing on SQLAlchemy errors using the DatasetSqlError exception

That's what Nikhil _was_ doing prior to this last round of changes, and [apparently I talked him out of it](https://github.com/distributed-system-analysis/pbench/pull/2440#discussion_r708466285). 🙂  (In my defense, he was having trouble making that exception object fit his case, and it was upsetting poor LGTM.)